target_chains/ethereum/contracts/contracts/pyth/Pyth.sol

```diff
@@ -432,84 +432,144 @@ abstract contract Pyth is
             }
 
             priceFeeds = new PythStructs.PriceFeed[](priceIds.length);
-
             for (uint i = 0; i < updateData.length; i++) {
-                bytes memory encoded;
-
-                {
-                    IWormhole.VM memory vm = parseAndVerifyBatchAttestationVM(
-                        updateData[i]
-                    );
-                    encoded = vm.payload;
-                }
+                if (
+                    updateData[i].length > 4 &&
+                    UnsafeBytesLib.toUint32(updateData[i], 0) ==
+                    ACCUMULATOR_MAGIC
+                ) {
+                    (
+                        PythInternalStructs.PriceInfo[]
+                            memory accumulatorPriceInfos,
+                        bytes32[] memory accumulatorPriceIds
+                    ) = parsePricesUsingAccumulator(updateData[i]);
```

Comment:
Let's keep in mind that using big memories has a lot of gas usage and in the future PRs we should fix it.